### PR TITLE
Refactor modules in many-rs and add support for async.status in abci …

### PR DESCRIPTION
**Pull request by hansl**
…frontend

This is the framework followup for https://github.com/l-1-labs/many-rs/pull/43

This will fail CI until the PR in many-rs is merged and I update the dependencies.

### CODE REVIEW

**Review by fmorency — CHANGES_REQUESTED** [review]
Not sure if this is expected/normal, but the `run.sh` script behavior is different between this version and `main`.

The output of the `tendermint-ledger` window progress **much slower** than before... I feel that everything is _slower_.

**Comment by hansl** [comment]
Yes I needed to slow down block production to test the async attribute. That's the `toml_set` changes you see. This PR doesn't work without the `many-rs` PR anyway so I'll fix it when the other one is merged.

**Comment by fmorency** [comment]
Let's aim to merge this today if possible, I need this to push my first `coset` draft.

File: src/many-kvstore/src/module.rs [hunk 1]
```diff
@@ -105,7 +103,9 @@ impl KvStoreModuleBackend for KvStoreModuleImpl {
             value: value.map(|x| x.into()),
         })
     }
+}
 
+impl KvStoreCommandsModuleBackend for KvStoreModuleImpl {
```
[the remark this answers is not in the document]
Comment (hansl) [reply src/many-kvstore/src/module.rs:108]: Done.

**Review by fmorency — APPROVED** [review]

**Pull request merged**